packages/render-fidelity-tools/test/renderers/model-viewer/index.html

```diff
@@ -58,7 +58,11 @@
 
       modelViewer.addEventListener('model-visibility', (event) => {
         if (event.detail.visible) {
-          self.dispatchEvent(new CustomEvent('model-ready'));
+          //TODO: this is a temporary workaround for taking screenshots before the model has been loaded
+          //This will be fixed when the screenshot function is refactored.
+          setTimeout(()=>{
+            self.dispatchEvent(new CustomEvent('model-ready'));
+          }, 300);
```

Comment:
Is 300ms good enough? Do you want it to be higher?

Reply:
Yeah, 300ms is the minimum, but you might want to put 500 just to give yourself some headroom. More importantly though, I think this timeout should be in your raf callback instead, as it can take some time to render that first frame and I think we want to wait after that.

Reply:
there isn't raf callback in the original script -- there isn't a model-viewer.ts file at all. I guess it's because model-viewer is already a web-component so there's we need to do with it as what we did for filament. Do you want me to add a raf call back for model-viewer?

Reply:
No problem; I forgot we do this internally in model-viewer already.